Lesson10/src/com/homework/shop/Shop.java

```diff
@@ -0,0 +1,32 @@
+package com.homework.shop;
+
+import java.util.List;
+
+public class Shop {
+    private final List<Cashbox> cashboxList;
+    private int freeCashbox;
+    private int countClients;
+
+
+    public Shop(List<Cashbox> cashboxList, int countClients) {
+        this.cashboxList = cashboxList;
+        freeCashbox = cashboxList.size();
+        this.countClients = countClients;
+    }
+
+    public synchronized void buy(String purchase) {
```

Comment:
so your buy method is synchronized which means that only 1 thread at a time can work here, which mean even if you have plenty of cash boxes, you will serve only 1 customer at a time